From: Tal Zussman <tz2294@columbia.edu>
Subject: mm: remove stray references to struct pagevec
Date: Wed, 25 Feb 2026 12:57:16 -0500

struct pagevec was removed in commit 1e0877d58b1e ("mm: remove struct
pagevec").  Remove remaining forward declarations and change
__folio_batch_release()'s declaration to match its definition.

Link: https://lkml.kernel.org/r/20260225-pagevec_cleanup-v1-1-38e2246363d2@columbia.edu
Signed-off-by: Tal Zussman <tz2294@columbia.edu>
Cc: Chao Yu <chao@kernel.org>
Cc: David Howells <dhowells@redhat.com>
Cc: Marc Dionne <marc.dionne@auristor.com>
Cc: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 fs/afs/internal.h       |    1 -
 fs/f2fs/f2fs.h          |    2 --
 include/linux/pagevec.h |    2 +-
 include/linux/swap.h    |    2 --
 4 files changed, 1 insertion(+), 6 deletions(-)

--- a/fs/afs/internal.h~mm-remove-stray-references-to-struct-pagevec
+++ a/fs/afs/internal.h
@@ -31,7 +31,6 @@
 
 #define AFS_CELL_MAX_ADDRS 15
 
-struct pagevec;
 struct afs_call;
 struct afs_vnode;
 struct afs_server_probe;
--- a/fs/f2fs/f2fs.h~mm-remove-stray-references-to-struct-pagevec
+++ a/fs/f2fs/f2fs.h
@@ -28,8 +28,6 @@
 #include <linux/fscrypt.h>
 #include <linux/fsverity.h>
 
-struct pagevec;
-
 #ifdef CONFIG_F2FS_CHECK_FS
 #define f2fs_bug_on(sbi, condition)	BUG_ON(condition)
 #else
--- a/include/linux/pagevec.h~mm-remove-stray-references-to-struct-pagevec
+++ a/include/linux/pagevec.h
@@ -93,7 +93,7 @@ static inline struct folio *folio_batch_
 	return fbatch->folios[fbatch->i++];
 }
 
-void __folio_batch_release(struct folio_batch *pvec);
+void __folio_batch_release(struct folio_batch *fbatch);
 
 static inline void folio_batch_release(struct folio_batch *fbatch)
 {
--- a/include/linux/swap.h~mm-remove-stray-references-to-struct-pagevec
+++ a/include/linux/swap.h
@@ -20,8 +20,6 @@ struct notifier_block;
 
 struct bio;
 
-struct pagevec;
-
 #define SWAP_FLAG_PREFER	0x8000	/* set if swap priority specified */
 #define SWAP_FLAG_PRIO_MASK	0x7fff
 #define SWAP_FLAG_DISCARD	0x10000 /* enable discard for swap */
_
